uni/lib/model/entities/library_occupation.dart

```diff
@@ -0,0 +1,47 @@
+/// Overall occupation of the library
+class LibraryOccupation {
+  int occupation = 0;
+  int capacity = 0;
+  List<FloorOccupation> floors = [];
+
+  LibraryOccupation(this.occupation, this.capacity);
+
+  void addFloor(FloorOccupation floor) {
+    floors.add(floor);
+    occupation += floor.occupation;
+    capacity += floor.capacity;
+  }
+
+  int get percentage {
+    if (capacity == 0) return 0;
+    return (occupation * 100 / capacity).round();
+  }
+
+  FloorOccupation getFloor(int number) {
+    if (floors.length < number || number < 0) return FloorOccupation(0, 0, 0);
+    return floors[number - 1];
+  }
+}
```

Comment:
Can we use the getter syntax here, such as in the method above?

Reply:
I don't think so, as Dart getters do not allow for parameters.